### ISSUE TITLE
Button with Icon does not obey Icon Spacing

### ISSUE THREAD

**Issue by arcfide** [IBM/carbon-components-svelte#174]
In 0.5.1 when a Button is used with text and an Icon, the Icon appears right next to the text, rather than the Icon being offset. 

```
<Button kind="primary" type="submit" icon={ArrowRight16}>
        Let's Play!
      </Button>
```

**Comment by metonym** [comment]
Thank you @arcfide for raising this issue.

In [Button.svelte](https://github.com/IBM/carbon-components-svelte/blob/master/src/components/Button/Button.svelte#L27), `hasIconOnly` incorrectly evaluates to `true` because `$$props.$$slots` is `undefined`.

**Comment by metonym** [comment]
Released in [v0.6.0](https://github.com/IBM/carbon-components-svelte/blob/master/CHANGELOG.md#060---2020-05-07)
